.gitignore

```diff
@@ -3,12 +3,19 @@
 .bundle
 build/*
 tools/chef/cookbooks/*
+tools/chef/site-cookbooks/**/cookbooks/*
 tools/chef/tmp/*
 tools/assets/*
 bin/*
 .DS_Store
 secret.key
 
+# Chef
+**/.kitchen/
+**/cucumber/Gemfile.lock
+tools/chef/site-cookbooks/*/Berksfile.lock
+tools/chef/.chef/*_encrypted_data_bag_secret
```

Comment:
Not sure what you mean by the last comment. What do you envisage?

Separate is more annoying for sure, but helps stop the scenario you depicted where a pipeline server is compromised and suddenly production passwords can be known, from the git clone on the same box.